x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureaistudio/AzureAiStudioEndpointType.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.services.azureaistudio;
+
+import org.elasticsearch.ElasticsearchStatusException;
+import org.elasticsearch.rest.RestStatus;
+
+import java.util.Locale;
+import java.util.Objects;
+
+public enum AzureAiStudioEndpointType {
+    TOKEN,
+    REALTIME;
+
+    public static String NAME = "azure_ai_studio_endpoint_type";
+
+    public static AzureAiStudioEndpointType fromString(String name) {
+        return valueOf(name.trim().toUpperCase(Locale.ROOT));
+    }
+
+    public static AzureAiStudioEndpointType fromStringOrStatusException(String name) {
```

Comment:
I believe this is unused, do we need it?

Reply:
Yep - I think this (and below) can be removed... I think I had put this in before I realized the enum validation already is in place (with it's own error format too)